ops/model.py

```diff
@@ -218,23 +219,22 @@ def __getitem__(self, relation_name):
         relation_list = self._data[relation_name]
         if relation_list is None:
             relation_list = self._data[relation_name] = []
-            relation_role = self._relations_meta[relation_name].role
+            our_app = self._our_unit.app if self._relations_meta[relation_name].role == 'peers' else None
             for relation_id in self._backend.relation_ids(relation_name):
-                relation_list.append(Relation(relation_name, relation_id, relation_role, self._our_unit, self._backend, self._cache))
+                relation_list.append(Relation(relation_name, relation_id, our_app, self._our_unit, self._backend, self._cache))
         return relation_list
 
 
 class Relation:
-    def __init__(self, relation_name, relation_id, role, our_unit, backend, cache):
+    def __init__(self, relation_name, relation_id, our_app, our_unit, backend, cache):
         self.name = relation_name
         self.id = relation_id
-        self.role = role
         self.app = None
         self.units = set()
 
         # For peer relations, both the remote and the local app are the same.
-        if self.role == 'peers':
-            self.app = our_unit.app
+        if our_app is not None:
+            self.app = our_app
```

Comment:
Instead of
```
self.app = None
(...)
if our_app is not None:
    self.app = our_app
```
This can be simply:
```
self.app = our_app
```
But! This shows a real problem in my recommendation: this is awkward because this is not our app. This is the relations app.. it makes no sense to tell the relation what the app is only sometimes.

Your suggestion of having an `is_peer` attribute seems more reasonable now. Sorry for taking you into this misleading path.

Reply:
Alternatively, we might have relation_app as a parameter, instead of our_app.